### ISSUE TITLE
[R/S] weird double conversions

### ISSUE THREAD

**Issue by hiiamboris** [red/red#4103]
**Describe the bug**

Double type conversions produce unexplainable results:
```
Red []
#system [
	p: as cell! 00100000h
	print-line ["p="p " p+1="p + 1]
	print-line ["(as byte-ptr! (as-integer p))=" (as byte-ptr! (as-integer p))]
	print-line ["(as byte-ptr! (as-integer (p + 1)))=" (as byte-ptr! (as-integer (p + 1)))]
	p: p + 1
	print-line ["p + 1 = " p]
	i: as-integer p
	print-line ["as-integer p + 1 = " i]
	b: as byte-ptr! i
	print-line ["as byte-ptr! as-integer p + 1 = " b]
]
```
Output:
```
p=00100000 p+1=00100010                     
(as byte-ptr! (as-integer p))=00100000      
(as byte-ptr! (as-integer (p + 1)))=0040E40A              ;) should be 00100010
p + 1 = 00100010                            
as-integer p + 1 = 1048592                  
as byte-ptr! as-integer p + 1 = 00100010    
```

**Platform version (please complete the following information)**
```
Red 0.6.4 for Windows built 18-Oct-2019/15:33:41+03:00 commit #ed913ef
```

**Comment by qtxie** [comment]
I think the compiler doesn't support it.

**Comment by hiiamboris** [comment]
Pity. It would've been helpful in macros, to silence compiler warnings that conversion from byte-ptr to byte-ptr is not necessary.
A workaround for now is: `as byte-ptr! 0 + as-integer p + 1`

**Comment by hiiamboris** [comment]
Looks like it's been there for 7 years https://github.com/red/Red/issues/142
Not sure if it was bugged like now or just not allowed

**Comment by dockimbel** [comment]
After the fix for #4414, I get:
```
p=00100000 p+1=00100010
(as byte-ptr! (as-integer p))=00100000
(as byte-ptr! (as-integer (p + 1)))=00100010
p + 1 = 00100010
as-integer p + 1 = 1048592
as byte-ptr! as-integer p + 1 = 00100010
```

So we can consider this wish granted, at least for pointer type castings.

**Comment by hiiamboris** [comment]
Original snippet and `pointer-test.reds` are both working for me in recent builds.
